lib/membrane/video_compositor/compile/download_compositor.ex

```diff
@@ -43,26 +43,46 @@ defmodule Mix.Tasks.Compile.DownloadCompositor do
 
   @spec system_architecture() :: String.t()
   defp system_architecture() do
-    case :os.type() do
-      {:unix, :darwin} ->
-        system_architecture = :erlang.system_info(:system_architecture) |> to_string()
+    system =
+      case :os.type() do
+        {:unix, :darwin} -> :darwin
+        {:unix, :linux} -> :linux
+        os_type -> os_type
+      end
 
-        cond do
-          Regex.match?(~r/aarch64/, system_architecture) ->
-            "darwin_aarch64"
+    arch = cpu_architecture()
 
-          Regex.match?(~r/x86_64/, system_architecture) ->
-            "darwin_x86_64"
+    case {system, arch} do
+      {:linux, :x86} ->
+        "linux_x86_64"
 
-          true ->
-            raise "Unsupported system architecture: #{system_architecture}"
-        end
+      {:linux, :aarch64} ->
+        "linux_aarch64"
 
-      {:unix, :linux} ->
-        "linux_x86_64"
+      {:darwin, :x86} ->
+        "darwin_x86_64"
+
+      {:darwin, :aarch64} ->
+        "darwin_aarch64"
+
+      {system, arch} ->
+        Membrane.Logger.warning("Unsupported platform: #{system} #{arch}")
+    end
```

Comment:
This doesn't match the function spec. Either change the spec and handle it there or raise it here (then you need to also add `no_return` to spec)